File: LABS/Lab17/with-prob.rb
def with_prob (prob)
  yield if (Random.rand < prob)
end

with_prob 0.42 do
  puts "There is a 42% chance that this code will print"
end



def foo x
  with_prob 0.5 do
    puts "Executing with_prob block"
    return 0
  end
  return x
end

puts "Foo is #{foo 1}"

File: LABS/Lab17/withProb.js
function withProb(prob, f) {
  if (Math.random() < prob) {
    return f();
  }
}


function foo(x) {
  withProb(0.5, function() {
    console.log("Execution withProb callback");
    return 0;
  });
  return x;
}

console.log("Foo is " + foo(1));

